plone/app/content/browser/selection.py

```diff
@@ -100,9 +100,12 @@ def get_selectable_items(self):
         view_types = registry.get("plone.types_use_view_action_in_listings", [])
         default_page_types = registry.get("plone.default_page_types", [])
         portal_types = getToolByName(self.context, "portal_types")
+        portal_catalog = getToolByName(self.context, "portal_catalog")
 
         results = []
-        for brain in context.getFolderContents():
+        for brain in portal_catalog(
+            path={"query": "/".join(context.getPhysicalPath()), "depth": 1}
```

Comment:
fwiw, you could use context.absolute_url_path() here

Reply:
The getFolderContents script also sorts by getObjPositionInParent by default

Reply:
thx, I updated it.